src/quicklookUI.cs

```diff
@@ -172,5 +173,69 @@ interface QLPreviewingController {
 		[Mac (10,15)]
 		[Export ("preparePreviewOfFileAtURL:completionHandler:")]
 		void PreparePreviewOfFile (NSUrl url, Action<NSError> completionHandler);
+
+		[iOS (15,0), Mac (12,0), MacCatalyst (15,0)]
+		[Export ("providePreviewForFileRequest:completionHandler:")]
+		void ProvidePreview (QLFilePreviewRequest request, Action<QLPreviewReply, NSError> handler);
+	}
+
+	[NoWatch, NoTV, Mac (12,0), iOS (15,0), MacCatalyst (15,0)]
+	[BaseType (typeof(NSObject))]
+	interface QLFilePreviewRequest
+	{
+		[Export ("fileURL")]
+		NSUrl FileUrl { get; }
+	}
+
+	[NoWatch, NoTV, Mac (12,0), iOS (15,0), MacCatalyst (15,0)]
+	[DisableDefaultCtor]
+	[BaseType (typeof(NSObject))]
+	interface QLPreviewProvider : NSExtensionRequestHandling
+	{
+	}
+
+	[NoWatch, NoTV, Mac (12,0), iOS (15,0), MacCatalyst (15,0)]
+	[BaseType (typeof(NSObject))]
+	interface QLPreviewReplyAttachment
+	{
+		[Export ("data")]
+		NSData Data { get; }
+
+		[Export ("contentType")]
+		UTType ContentType { get; }
+
+		[Export ("initWithData:contentType:")]
+		IntPtr Constructor (NSData data, UTType contentType);
+	}
+
+	delegate bool QLPreviewReplyDrawingHandler (CGContext context, QLPreviewReply reply, out NSError error);
+	delegate NSData QLPreviewReplyDataCreationHandler (QLPreviewReply reply, out NSError error);
+	delegate CGPDFDocument QLPreviewReplyUIDocumentCreationHandler (QLPreviewReply reply, out NSError error);
+
+	[NoWatch, NoTV, Mac (12,0), iOS (15,0), MacCatalyst (15,0)]
+	[BaseType (typeof(NSObject))]
+	interface QLPreviewReply
+	{
+		[Export ("stringEncoding")]
+		nuint StringEncoding { get; set; }
```

Comment:
from wiki / diff
```
+/** String encoding for text or html based previews. Defaults to NSUTF8StringEncoding. */
+@property NSStringEncoding stringEncoding;
```

there's already an `enum` for this, named `NSStringEncoding`
see `src/Foundation/Enum.cs`